javascript/ql/src/Security/CWE-830/FunctionalityFromUntrustedSource.ql

```diff
@@ -0,0 +1,181 @@
+/**
+ * @name Inclusion of untrusted functionality by a HTML element.
+ * @description Including untrusted functionality by a HTML element
+ *              opens up for potential man-in-the-middle attacks.
+ * @kind problem
+ * @problem.severity warning
+ * @security-severity 8.1
+ * @precision high
+ * @id js/functionality-from-untrusted-source
+ * @tags security
+ *       external/cwe/cwe-830
+ */
+
+import javascript
+import semmle.javascript.HTML
+import semmle.javascript.dataflow.TaintTracking
+
+module Generic {
+  /** A `CallNode` that creates an element of kind `name`. */
+  predicate isCreateElementNode(DataFlow::CallNode call, string name) {
+    call = DataFlow::globalVarRef("document").getAMethodCall("createElement") and
+    call.getArgument(0).getStringValue().toLowerCase() = name
+  }
+
+  /**
+   * True if `rhs` is being assigned to the `propName` property of an HTML
+   * element created by `createElementCall`.
+   */
+  predicate isPropWrite(DataFlow::CallNode createElementCall, string propName, DataFlow::Node rhs) {
```

Comment:
We have a predicate for doing this: `SourceNode.getAPropertyWrite`. It is available on `CallNode` as it is a subclass of `SourceNode`. I'd suggest using that instead (see inline suggestions below).

Reply:
oh, I had missed that. THX :)